### ISSUE TITLE
Make the constructor of ai.djl.huggingface.tokenizers.Encoding  public.

### ISSUE THREAD

**Issue by demq** [deepjavalibrary/djl#1938]
## Description

The constructor of ai.djl.huggingface.tokenizers.Encoding is currently package-private, which means we cannot create new instances "from scratch" in external user packages. 

Will this change the current api? How? - No.

Who will benefit from this enhancement? - Users working with HuggingFaceTokenizer.

**Comment by frankfliu** [comment]
@demq 
The `Encoding` object is created by HuggingfaceTokenizer, can you provide an use case why you need create a `Encoding` object yourself?

**Comment by demq** [comment]
For example I want to create a subclass that holds some extra information, such as the offset of the context tokens in QA applications, this helps to optimise the postProcessing step of a custom QuestionAnsweringTranslator (you don't want to consider the "undersired" question and separator tokens when computing softmax of the logits.). This subclass can then also be used to hold the results of custom tokenziers, these tokenizers can be used interchangeably with the HuggingFace ones in projects where the models are configurable.

It is trivial to write a "clone" class of Encoding and initalize it with an instance of Encoding, I just thought creating a subclass might be a cleaner and useful to others as well.

**Comment by frankfliu** [comment]
@demq 
That make sense. We can make the constructor protected. Would you mind create a PR?

**Comment by demq** [comment]
Done: https://github.com/deepjavalibrary/djl/pull/1945#issue-1348722138
